### PR TITLE
Fix modal state activation

### PR DESCRIPTION
**Pull request by nikvoloshin**
Fixes a bug due to which modal state itself cannot be activated from its subtree.

### CODE REVIEW

File: core/src/test/kotlin/com/justai/jaicf/core/test/scenario/ScdnarioModalityTest.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package com.justai.jaicf.core.test.scenario
```
Comment (Denire) [inline core/src/test/kotlin/com/justai/jaicf/core/test/scenario/ScdnarioModalityTest.kt:1]: This file has typo error

Comment (nikvoloshin) [reply core/src/test/kotlin/com/justai/jaicf/core/test/scenario/ScdnarioModalityTest.kt:1]: Oops

File: core/src/main/kotlin/com/justai/jaicf/activator/BaseActivator.kt [hunk 2]
```diff
@@ -94,6 +95,12 @@ abstract class BaseActivator(private val model: ScenarioModel) : Activator {
         }
 
         val transitionsMap = model.transitions.groupBy { it.fromState }
-        return availableStates.flatMap { transitionsMap[it] ?: emptyList() }
+
+        val lastState = StatePath.parse(availableStates.last())
```
Comment (Denire) [inline core/src/main/kotlin/com/justai/jaicf/activator/BaseActivator.kt:99]: Isn't it true that this lastState is always equal to currentState?

Comment (nikvoloshin) [reply core/src/main/kotlin/com/justai/jaicf/activator/BaseActivator.kt:99]: No, we process states in reverse order from current state:
Suppose currentState = /a/b/c, then the order is following: /a/b/c -> /a/b -> /a -> /
So, lastState - is either first modal ancestor, or root

**Review by Denire — APPROVED** [review]

File: core/src/main/kotlin/com/justai/jaicf/activator/BaseActivator.kt [hunk 3]
```diff
@@ -94,6 +95,12 @@ abstract class BaseActivator(private val model: ScenarioModel) : Activator {
         }
 
         val transitionsMap = model.transitions.groupBy { it.fromState }
```
Comment (veptechno) [inline core/src/main/kotlin/com/justai/jaicf/activator/BaseActivator.kt:97]: Do you want to simplify the method?

**Review by morfeusys — APPROVED** [review]

**Pull request merged**